packages/ckeditor5-fullscreen/src/handlers/abstracteditor.ts

```diff
@@ -177,4 +180,94 @@ export default class AbstractEditorHandler {
+	 * utilizing the empty space on the right side of the editable element.
+	 */
+	public registerFullscreenDialogPositionAdjustements(): void {
+		if ( !this._editor.plugins.has( 'Dialog' ) ) {
+			return;
+		}
+
+		const dialog = this._editor.plugins.get( 'Dialog' ) as Dialog;
+
+		this.setNewDialogPosition();
+
+		dialog.on( 'change:isOpen', this.updateDialogPositionCallback, { priority: 'highest' } );
+	}
+
+	/**
+	 * Removes an event listener that adjusts the dialog's position in fullscreen mode.
+	 */
+	public unregisterFullscreenDialogPositionAdjustements(): void {
+		if ( !this._editor.plugins.has( 'Dialog' ) ) {
+			return;
+		}
+
+		const dialog = this._editor.plugins.get( 'Dialog' ) as Dialog;
+		const dialogView = dialog.view;
+
+		if ( dialogView && dialogView.position === null ) {
+			dialogView.position = DialogViewPosition.EDITOR_TOP_SIDE;
+		}
+
+		dialogView?.updatePosition();
+
+		dialog.off( 'change:isOpen', this.updateDialogPositionCallback );
+	}
+
+	public updateDialogPositionCallback = this.updateDialogPosition.bind( this );
+
+	/**
+	 * An event triggered on dialog opening that sets a new position or restores previous values.
+	 */
+	private updateDialogPosition( _evt: EventInfo, _name: string, isOpen: boolean ): void {
+		if ( isOpen ) {
+			this.setNewDialogPosition();
+		} else {
+			const dialog = this._editor.plugins.get( 'Dialog' ) as Dialog;
+			const dialogView = dialog.view;
+
+			if ( dialogView?.position === null ) {
+				dialogView.position = DialogViewPosition.EDITOR_TOP_SIDE;
+			}
+		}
+	}
+
+	/**
+	 * Adjusts the dialog position to utilize the empty space on the right side of the editable.
+	 * The new dialog position should be on the right side of the fullscreen view with a 30px margin.
+	 * Only dialogs with the position set to "editor-top-side" should have their position changed.
+	 */
+	public setNewDialogPosition(): void {
+		if ( !this._editor.plugins.has( 'Dialog' ) ) {
+			return;
+		}
+
+		const dialog = this._editor.plugins.get( 'Dialog' ) as Dialog;
+		const dialogView = dialog.view!;
+
+		if ( !dialogView || dialogView.position !== DialogViewPosition.EDITOR_TOP_SIDE ) {
+			return;
+		}
+
+		const fullscreenViewContainerRect = this._getVisibleContainerRect( this._container! );
+		const editorContainerRect = this._getVisibleContainerRect( document.querySelector( '.ck-fullscreen__editor' )! );
+		const dialogRect = this._getVisibleContainerRect( dialogView.element!.querySelector( '.ck-dialog' ) as HTMLElement );
+
+		if ( fullscreenViewContainerRect && editorContainerRect && dialogRect ) {
+			const DIALOG_OFFSET = 30;
+			dialogView.position = null;
+
+			dialogView.moveTo(
+				fullscreenViewContainerRect.left + fullscreenViewContainerRect.width - dialogRect.width - DIALOG_OFFSET,
+				editorContainerRect.top + DIALOG_OFFSET
+			);
+		}
+	}
+
+	private _getVisibleContainerRect( container: HTMLElement ): Rect | null {
+		return new Rect( container )?.getVisible();
+	}
```

Comment:
This helper is only used in one method so maybe we'd move it to that functions scope to keep the interface cleaner. Or leave it here but add docs.